plutus-tx-plugin/test/IsData/Spec.hs

```diff
@@ -158,6 +164,7 @@ tests =
       , goldenUEval "mono" [plc (Proxy @"mono") (isDataRoundtrip (Mono2 2))]
       , goldenUEval "poly" [plc (Proxy @"poly") (isDataRoundtrip (Poly1 (1 :: Integer) (2 :: Integer)))]
       , goldenUEval "record" [plc (Proxy @"record") (isDataRoundtrip (MyMonoRecord 1 2))]
+      , goldenUEval "recordAsList" [plc (Proxy @"record") (isDataRoundtrip (MyMonoRecordAsList 1 2))]
```

Comment:
Need something better than this since `goldenUEval` doesn't check if given value actually is `True` value.

Reply:
Is your concern that `constr 0 []` could represent something else other than True? True, but highly unlikely.

Reply:
It should be okay for now, but we've seen representation of Bool changing(with introduction of SOP) and also more importantly, there's no reason to have this as golden at all! Golden from these cases are just purely unnecessary noises 

I'll make some changes